package com.magneto;

import net.minidev.json.parser.JSONParser;
import net.minidev.json.parser.ParseException;
import org.junit.Test;
import static org.junit.Assert.assertTrue;
import static org.junit.Assert.fail;

public class CVE_2021_27568_Testcase1 {

    @Test(timeout = 60000)
    public void testInvalidNumbers() {
        JSONParser p = new JSONParser();
        String invalidNum = "2e+";
        String json = "{v:" + invalidNum + "}";
        try {
            p.parse(json);
            fail("the test fail");
        } catch (Exception e) {
            // everything is fine, we expected ParseException
            validateThrow(e);
        }
    }

    public void validateThrow(Throwable e) {
        assertTrue(e instanceof ParseException);
    }
}
